lms/djangoapps/instructor/lti_connection.py

```diff
@@ -0,0 +1,140 @@
+    if len(grade_row) >= 6:
+        comment = grade_row[5]
+    if anon_id == '':
+        raise LTIConnectionError(
+            _("GRADE POST FAILED: user id {user_id} (email: {email}) is missing a proper anonymized id").format(
+                user_id=uid,
+                email=email,
+            )
+        )
+
+    url = url_base + anon_id
+
+    payload = {
+        '@context': 'http://purl.imsglobal.org/ctx/lis/v2/Result',
+        '@type': 'Result',
+        'comment': comment,
+        'resultScore': grade / total,
+    }
+
+    resp = _send_lti_2_json_request(
+        'PUT',
+        url,
+        key,
+        secret,
+        data=json.dumps(payload),
+    )
+    if resp.status_code != 200:
+        raise LTIConnectionError(
+            _("GRADE POST FAILED: user id {user_id} (email: {email}). Request was unsuccessful.").format(
+                user_id=uid,
+                email=email,
+            )
+        )
+    else:
+        success_msg = _("GRADE POST SUCCESSFUL: user id {user_id} (email: {email}).").format(
+            user_id=uid,
+            email=email,
+        )
+        log.info(success_msg)
+        return success_msg
+
+
+def _send_lti_2_json_request(method, url, key, secret, data=None):
+    """
+    Issue a session-based LTI request
+    """
+    session = requests.Session()
+    request = requests.Request(method, url, data=data)
+    request.headers.update({
+        'Content-Type': 'application/vnd.ims.lis.v2.result+json',
+    })
+    request_prepared = session.prepare_request(request)
+    auth_header_val = _get_authorization_header(request_prepared, key, secret)
+    request.headers.update({
+        'Authorization': auth_header_val,
+    })
+    request_prepared_authorized = session.prepare_request(request)
+    response = session.send(request_prepared_authorized)
+    return response
+
+
+def _get_authorization_header(request, client_key, client_secret):
+    """
+    Get proper HTTP Authorization header for a given request
+
+    Arguments:
+        request: Request object to log Authorization header for
+
+    Returns:
+        authorization header
+    """
+    sha1 = hashlib.sha1()
+    body = request.body or ''
+    sha1.update(body)
+    oauth_body_hash = unicode(base64.b64encode(
+        sha1.digest()  # pylint: disable=too-many-function-args
+    ))
+    client = Client(client_key, client_secret)
+    params = client.get_oauth_params(None)
+    params.append((u'oauth_body_hash', oauth_body_hash))
+    mock_request = mock.Mock(
+        uri=unicode(urllib.unquote(request.url)),
+        headers=request.headers,
+        body=u'',
+        decoded_body=u'',
+        oauth_params=params,
+        http_method=unicode(request.method),
+    )
+    sig = client.get_oauth_signature(mock_request)
+    mock_request.oauth_params.append((u'oauth_signature', sig))
+
+    headers = client._render(  # pylint: disable=protected-access
+        mock_request
+    )[1]
```

Comment:
As per the documentation, client._render does the following:
Returns a 3-tuple containing the request URI, headers, and body.